os/storage/cfs/cfs-coffee.c

```diff
@@ -75,6 +75,10 @@
 #error "Cannot have COFFEE_APPEND_ONLY set when COFFEE_MICRO_LOGS is set."
 #endif
 
+#if COFFEE_NAME_LENGTH != CFS_DIR_ENTRY_NAME_LENGTH
```

Comment:
why not simply use `COFFEE_NAME_LENGTH` instead of introducing the new constant `CFS_DIR_ENTRY_NAME_LENGTH`?

Reply:
That would create a dependency from CFS to Coffee, which I assume would not survive sunlight since Coffee implements CFS(?).

However, I see this break at least z1 and sky. Leaving out the constant and add only the memcpy()-change is an option. Users would have to be aware that their directory-names would be limited by `COFFEE_NAME_LENGTH` and not the hard-coded `32` in cfs.h. With that said I notice this comment in coffee: 
```
  /*
   * Coffee is only guaranteed to support the directory names "/" and ".",
   * but it does not enforce this currently.
   */
```
So, it seems the directory-naming-support is dubious already. Leave as-is and fix the memcpy?

Reply:
The warning could be removed, and then just keep the fix of the memcpy call to the shorter name. Additionally, record->name should be null-terminated with an offset that is the shorter of the two lengths.

Reply:
:+1:

> Additionally, record->name should be null-terminated with an offset that is the shorter of the two lengths.

Indeed! fixed